### PR TITLE
[fix](catalog recycle bin) table partition meta is error if recover partition in some case

### PR DESCRIPTION
**Pull request by mymeiyi**
## Proposed changes

If recover partition use a new partition name:
```
recover partition p3 as p6 from db2.tbl1;
```

if the table contains a partition `p3`, the table partition meta data `idToPartition` and `nameToPartition` is inconsistent: 
p2

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java [hunk 1]
```diff
@@ -811,18 +810,18 @@ public synchronized void recoverPartition(long dbId, OlapTable table, String par
...
 
-        // recover partition
+        // check if partition name exists
         Partition recoverPartition = recoverPartitionInfo.getPartition();
         Preconditions.checkState(recoverPartition.getName().equalsIgnoreCase(partitionName));
         if (!Strings.isNullOrEmpty(newPartitionName)) {
             if (table.checkPartitionNameExist(newPartitionName)) {
                 throw new DdlException("Partition name[" + newPartitionName + "] is already used");
             }
+            recoverPartition.setName(newPartitionName);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java:820]: before set new name,   check if table contains other partition with the same old partition name.  if not contains old partition name,  remove old partition name from table's nameToPartition ?

**Review by yujun777 — COMMENTED** [review]
LGTM

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by w41ter — APPROVED** [review]

**Pull request merged**